source/grid/grid_generator.cc

```diff
@@ -28,11 +29,1219 @@
+         * @return output Scaled vector with desired length.
+         * */
+        static std::vector<Point<2>>
+        set_airfoil_length(const std::vector<Point<2>> &input,
+                           const double                 desired_len)
+        {
+          std::vector<Point<2>> output(input.size());
+
+          const auto scale = desired_len / input.front().distance(input.back());
+
+          for (unsigned i = 0; i < input.size(); i++)
+            output[i] = input[i] * scale;
+
+          return output;
+        }
+
+        /**
+         * Interpolation among non_equidistant_points in order to obtain
+         * (number_points)-equidistant points
+         *
+         * @param[in] non_equidistan_points Vector containing non equidistan
+         * points.
+         * @param[in] number_points Indicating desired amount of equidistant
+         * points.
+         * @return equidist Vector containing (number_points) equidistant points.
+         */
+        static std::vector<Point<2>>
+        make_points_equidistant(
+          const std::vector<Point<2>> &non_equidistant_points,
+          const unsigned int           number_points)
+        {
+          const unsigned int n_points =
+            non_equidistant_points
+              .size(); // number provided airfoilpoints to interpolate
+
+          // calculate arclength
+          std::vector<double> arclength_L(non_equidistant_points.size(), 0);
+          for (unsigned int i = 0; i < non_equidistant_points.size() - 1; i++)
+            arclength_L[i + 1] =
+              arclength_L[i] +
+              non_equidistant_points[i + 1].distance(non_equidistant_points[i]);
+
+
+          const auto airfoil_length =
+            arclength_L.back(); // arclength upper or lower side
+          const auto deltaX = airfoil_length / (number_points - 1);
+
+          // Create equidistant points: keep the first (and last) point
+          // unchanged
+          std::vector<Point<2>> equidist(
+            number_points); // number_points is required points on each side for
+                            // mesh
+          equidist[0]                 = non_equidistant_points[0];
+          equidist[number_points - 1] = non_equidistant_points[n_points - 1];
+
+
+          // loop over all subsections
+          for (unsigned int j = 0, i = 1; j < n_points - 1; j++)
+            {
+              // get reference left and right end of this section
+              const auto Lj  = arclength_L[j];
+              const auto Ljp = arclength_L[j + 1];
+
+              while (Lj <= i * deltaX && i * deltaX <= Ljp &&
+                     i < number_points - 1)
+                {
+                  equidist[i] = Point<2>((i * deltaX - Lj) / (Ljp - Lj) *
+                                           (non_equidistant_points[j + 1] -
+                                            non_equidistant_points[j]) +
+                                         non_equidistant_points[j]);
+                  i++;
```

Comment:
We generally prefer the prefix form:
```suggestion
                  ++i;
```